### PR TITLE
test(comments): add more tests for PATCH/comments

### LINKED ISSUE #89: PATCH/ comments - add more test cases
Completing patch-comments tests with other possible API answers for this endpoint, listed in the GAD application documentation.

## ✏️ Description

-add test: 401Unauthorized -> when invalid token is sent in headers

-add test: 400 Bad Request -> when malformed json is used

-add test: 422 Unprocessable Entity -> when keys are missing and values are too long

-add test 404 Comment not found 

- [x] I created a descriptive title for this issue
- [x] I checked that it hasn't been reported yet
- [x] I assigned the issue or added `help-wanted` label
- [x] I added appropriate labels

### PR DESCRIPTION
**Pull request by Justyna-KO**
### Scope of changes:

-add test: 401Unauthorized -> when invalid token is sent in headers

-add test: 400 Bad Request -> when malformed json is used

-add test: 422 Unprocessable Entity -> when keys are missing and values are too long

-add test 404 Comment not found

## Any other comments?

Before submitting Pull Request, I've ensured that:

- [x] The PR is associated with the relevant Issue before its creation.
- [x] All new and existing tests passed
- [x] My commit messages follow the `kawqa-gad-playwright` project's [git commit message format](https://github.com/kat-kan/kawqa-gad-playwright/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: tests/api/comments/comments-patch.spec.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    const body = await response.json();
+    expect(body.error.message).toBe(expectedErrorMessage);
+  });
+
+  test('Returns 400 Bad Request status code when trying to update existing comments with malformed JSON', async ({
+    request,
+  }) => {
+    // Given
+    const malformedJson: string = `{
+      "article_id": 1,
```
Comment (Slawomir-DKl) [inline tests/api/comments/comments-patch.spec.ts:107]: use the same indent as in other elements of this JSON string

File: tests/api/comments/comments-patch.spec.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      'One of field is invalid (empty, invalid or too long) or there are some additional fields: Field validation: \"date\" has invalid format!';
+    // When
+    const response: APIResponse = await request.patch(
+      `${comments}/${setNewCommentId}`,
+      {
+        headers: setHeadersForRegularUser,
+        data: {
+          article_id: 1,
+          user_id: testUsers.regularUser.id,
+          body: `${newComment}`,
```
Comment (Slawomir-DKl) [inline tests/api/comments/comments-patch.spec.ts:137]: `body: newComment` should be sufficient - you don't concatenate any other variable to use backticks

**Review by Slawomir-DKl — CHANGES_REQUESTED** [review]
Please treat my comments as a beginner's suggestions/questions only :)

**Comment by Justyna-KO** [comment]
> Please treat my comments as a beginner's suggestions/questions only :)

thank you, very useful comments.
please review again.

**Review by Slawomir-DKl — APPROVED** [review]
Thank you for your changes. In fact, one test fails at the moment, but it comes from the future date in createNewComment() function. I've fixed it in https://github.com/kat-kan/kawqa-gad-playwright/pull/96 (I'm waiting for the Katarzyna's approval of that PR and I expect some nice conflicts, so please merge this PR to main first :) ).

File: tests/api/comments/comments-patch.spec.ts [hunk 3]
```diff
@@ -1,18 +1,23 @@
...
 import { APIResponse, expect, request, test } from '@playwright/test';
 import { customDate } from 'test-data/shared/date.generator';
 
 test.describe('PATCH comments/{id} endpoint tests', async () => {
   const comments: string = `/api/comments`;
-  let setHeadersForRegularUser;
-  let setNewCommentId;
-  const newComment = 'I changed my mind';
+  let setHeadersForRegularUser: { [key: string]: string };
+  let setInvalidHeaders: { [key: string]: string };
```
Comment (Slawomir-DKl) [inline tests/api/comments/comments-patch.spec.ts:13]: This variable is used only in one negative test, therefore I suggest removing its declaration and initialization from `describe` and `beforeAll` and entering `const setInvalidHeaders = await createInvalidHeaders();` into the test "Returns 401 Unauthorized status code when trying to update an existing comments with random text as token"

**Review by Slawomir-DKl — CHANGES_REQUESTED** [review]
I added one suggestion only.

**Review by Slawomir-DKl — APPROVED** [review]
Thank you for your changes.

**Pull request merged**